internal/server/server.go

```diff
@@ -0,0 +1,40 @@
+package server
+
+import (
+	"net/http"
+
+	"github.com/Pigmice2733/peregrine-backend/internal/config"
+	ihttp "github.com/Pigmice2733/peregrine-backend/internal/http"
+	routes "github.com/Pigmice2733/peregrine-backend/internal/routes/v1"
+	"github.com/gorilla/mux"
+)
+
+// Server holds information neccesary for the peregrine-backend server.
+type Server struct {
+	router *mux.Router
+	addr   string
+	origin string
+}
+
+// New returns a new peregrine server.
+func New(c config.Config) *Server {
+	s := &Server{addr: c.Server.Listen, origin: c.Server.Origin}
+
+	r := mux.NewRouter()
+	initRoutes(r)
+	s.router = r
+
+	return s
+}
+
+// Run starts serving at the given address.
+func (s *Server) Run() error {
+	return http.ListenAndServe(s.addr, s.router)
+}
+
+func initRoutes(r *mux.Router) {
+	r = r.PathPrefix("/v1").Subrouter()
```

Comment:
So we are versioning the API? Would it be appropriate to use SemVer, or at least have major and minor versions (minor = add stuff, major = remove stuff/change stuff)?

Or would it be better to just add stuff without changing the version, and only use the major version for removing stuff?

Reply:
I think just major version would be best, there's no need for the frontend to access a different version of the api as long as everything is backwards compatible.

Reply:
^